packages/puppeteer-core/src/api/Frame.ts

```diff
@@ -0,0 +1,965 @@
+   */
+  async tap(selector: string): Promise<void>;
+  async tap(): Promise<void> {
+    throw new Error('Not implemented');
+  }
+
+  /**
+   * Sends a `keydown`, `keypress`/`input`, and `keyup` event for each character
+   * in the text.
+   *
+   * @remarks
+   * To press a special key, like `Control` or `ArrowDown`, use
+   * {@link Keyboard.press}.
+   *
+   * @example
+   *
+   * ```ts
+   * await frame.type('#mytextarea', 'Hello'); // Types instantly
+   * await frame.type('#mytextarea', 'World', {delay: 100}); // Types slower, like a user
+   * ```
+   *
+   * @param selector - the selector for the element to type into. If there are
+   * multiple the first will be used.
+   * @param text - text to type into the element
+   * @param options - takes one option, `delay`, which sets the time to wait
+   * between key presses in milliseconds. Defaults to `0`.
+   */
+  async type(
+    selector: string,
+    text: string,
+    options?: {delay: number}
+  ): Promise<void>;
+  async type(): Promise<void> {
+    throw new Error('Not implemented');
+  }
+
+  /**
+   * @deprecated Replace with `new Promise(r => setTimeout(r, milliseconds));`.
+   *
+   * Causes your script to wait for the given number of milliseconds.
+   *
+   * @remarks
+   * It's generally recommended to not wait for a number of seconds, but instead
+   * use {@link Frame.waitForSelector}, {@link Frame.waitForXPath} or
+   * {@link Frame.waitForFunction} to wait for exactly the conditions you want.
+   *
+   * @example
+   *
+   * Wait for 1 second:
+   *
+   * ```ts
+   * await frame.waitForTimeout(1000);
+   * ```
+   *
+   * @param milliseconds - the number of milliseconds to wait.
+   */
+  waitForTimeout(milliseconds: number): Promise<void> {
+    return new Promise(resolve => {
+      setTimeout(resolve, milliseconds);
+    });
+  }
+
+  /**
+   * The frame's title.
+   */
+  async title(): Promise<string> {
+    throw new Error('Not implemented');
+  }
+
+  /**
+   * @internal
+   */
+  _deviceRequestPromptManager(): DeviceRequestPromptManager {
+    throw new Error('Not implemented');
+  }
+
+  /**
+   * This method is typically coupled with an action that triggers a device
+   * request from an api such as WebBluetooth.
+   *
+   * :::caution
+   *
+   * This must be called before the device request is made. It will not return a
+   * currently active device prompt.
+   *
+   * :::
+   *
+   * @example
+   *
+   * ```ts
+   * const [devicePrompt] = Promise.all([
+   *   frame.waitForDevicePrompt(),
+   *   frame.click('#connect-bluetooth'),
+   * ]);
+   * await devicePrompt.select(
+   *   await devicePrompt.waitForDevice(({name}) => name.includes('My Device'))
+   * );
+   * ```
+   */
+  waitForDevicePrompt(
+    options: WaitTimeoutOptions
+  ): Promise<DeviceRequestPrompt>;
+  waitForDevicePrompt(): Promise<DeviceRequestPrompt> {
+    throw new Error('Not implemented');
+  }
+
+  /**
+   * @internal
+   */
+  _navigated(
```

Comment:
do we need the methods here and below here?

Reply:
I will double check, I remember adding this to avoid some typescript fails, I think i can retype some places with the correct CDP Frame, and avoid this.